### PR TITLE
LPD-27880 Delete DatasetViewDateFilters.tescase

### PR DESCRIPTION
**Pull request by dsanz**
In this PR we are improving PW coverage of Date Filters in DSM and getting rid of old poshi tests related to this part of the DSM.

Reference:  https://liferay.atlassian.net/browse/LPD-27691

@markocikos @chestofwonders for the Refactor PR, Miguel is working on tests that should leverage changes I'm adding here (pls see `FiltersPage.ts`)

### CODE REVIEW

**Comment by dsanz** [comment]
ci:test:playwright

**Comment by dsanz** [comment]
ci:test:sf

File: modules/test/playwright/tests/frontend-data-set-views-web/dateRangeFilters.spec.ts [hunk 1]
```diff
@@ -27,6 +27,8 @@ export const dsmTest = mergeTests(
 
 let dataSetERC: string;
 let dataSetLabel: string;
+const NAME_COLUMN_INDEX = 1;
+const dateFieldName = 'dateCreated';
```
Comment (markocikos) [inline modules/test/playwright/tests/frontend-data-set-views-web/dateRangeFilters.spec.ts:31]: [nit] If it's a global constant, this should be `DATE_FIELD_NAME`,

File: modules/test/playwright/tests/frontend-data-set-views-web/dateRangeFilters.spec.ts [hunk 2]
```diff
@@ -121,24 +339,17 @@ fragmentTest(
 		});
 
 		const activeFilterButton = fdsFragmentPage.page.getByRole('button', {
-			name: `${filterLabel}: 1/1/2020 - 1/2/3020`,
+			name: `${filterLabel}:`,
```
Comment (markocikos) [inline modules/test/playwright/tests/frontend-data-set-views-web/dateRangeFilters.spec.ts:342]: Any reason to remove more specific locator?

File: modules/test/playwright/tests/frontend-data-set-views-web/pages/data_set/tabs/FiltersPage.ts [hunk 3]
```diff
@@ -67,6 +99,54 @@ export class FiltersPage {
...
+
+		const dateFilterOption = this.page.getByRole('option', {
+			name: filterBy,
+		});
+
+		await dateFilterOption.click();
+
+		from && await this.newDateRangeFilterModal.fromInput.fill(from);
+
+		to && await this.newDateRangeFilterModal.toInput.fill(to);
```
Comment (markocikos) [inline modules/test/playwright/tests/frontend-data-set-views-web/pages/data_set/tabs/FiltersPage.ts:145]: It looks odd to have an expression disguising as a statement. It technically works, but it doesn't _feel_ right. It makes me think of programming patterns I've seen in Perl, that take advantage of the fact that functions there return last evaluated value :smile: 

`if(from) {await ...}` is much cleaner.

**Pull request closed without merging**